### DISCUSSION TITLE
Exporting SQL statement metrics to Prometheus

### DISCUSSION THREAD

**Discussion by varun-gotgit** [inspectIT/inspectit-ocelot#1220] [Q&A]
I am exporting the metrics from the agent to prometheus. I can't seem to export metrics on SQL statements. I can see the DB connections and the total time taken for all the connections, but don't see any SQL statements. With pooling of JDBC connections, that doesn't give me any metrics on SQL statement execution time.

I can see the SQL Statements when exported to Jaeger though. Is there any way to export it to Prometheus?

**Comment by mariusoe** [comment]
Hi @varun-gotgit 

yes, by default there is no dedicated metric which represents the SQL statement duration based on executed SQL statement.

However, the agent exports the following metrics `service/out/count` and `service/out/responsetime/sum` by default  which also include the JDBC statistics, thus, the SQL execution time. You can filter these metrics using the tag `protocol = 'jdbc'` in order to get the execution time. Furthermore, you can split these data by the tag `target_external` in order to get the execution times by used database. 

If you need the execution times split by SQL statement, you have to create a custom rule and metric for collecting and exporting this data.

**Comment by varun-gotgit** [comment]
Hi mariusoe,

Thanks for clarifying.

Could you give me some examples of doing something like this with the exporter?

Thanks
Varun

**Comment by varun-gotgit** [comment]
@mariusoe,

Could you please help me with any possible documentation on the Prometheus exporter to ad the SQL statements?

I know that the agent is scraping the SQL statements because I can see them in Jaeger. So it's just a matter of adding it in  the prometheus exporter code.

Thanks in advance!

**Comment by mariusoe** [comment]
Hi @varun-gotgit 

yes, we'll put a new section into the documentation on how such a metric can be created. I think a detailed guide for this would help many users.
